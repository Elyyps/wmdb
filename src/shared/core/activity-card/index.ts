export * from './activity-card.component'