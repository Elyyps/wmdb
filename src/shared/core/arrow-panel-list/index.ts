export * from "./arrow-panel-list.component";
