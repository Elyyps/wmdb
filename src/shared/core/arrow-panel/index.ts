export * from "./arrow-panel.component";
