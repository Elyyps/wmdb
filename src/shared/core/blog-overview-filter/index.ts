export * from "./blog-overview-filter.component";
