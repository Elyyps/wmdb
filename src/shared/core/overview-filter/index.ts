export * from "./overview-filter.component";
