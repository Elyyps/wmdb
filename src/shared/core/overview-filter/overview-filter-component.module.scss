.button-search {
  cursor: pointer;
  background-color: $global-background;
  height: 45px;
  width: 100%;
  font-size: 16px;
  padding: 10px 32px 10px 16px;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid $gray-lighter;
  font-family: $global-font-family;
  display: flex;
  padding-left: 48px;
  position: relative;
  color: #6a6a6a;
  svg {
    position: absolute;
    left: 16px;
    top: 50%;
    transform: translateY(-50%);
  }
}
.divider {
  &::before {
    border-top: $gray-lighter 1px solid;
    display: block;
    margin: 12px 0 30px 0;
    content: "";
    @media (max-width: $breakpoint-small-max) {
      display: none;
    }
  }
}
.overview-filter-wrapper {
  background-color: $global-inverse-color;
}
.overview-filter {
  @media (max-width: $breakpoint-small-max) {
    padding: 16px;
    display: none;
  }
  &__header {
    padding-bottom: 12px;
    > * {
      margin: 8px 0 !important;
    }
  }
  &__title {
    button {
      background-color: transparent;
      border: none;
      cursor: pointer;
    }
    @media (max-width: $breakpoint-small-max) {
      display: flex;
      align-items: center;

      justify-content: space-between;
      font-size: 14px;
      font-weight: 600;
      color: $brand-primary;
      padding: 16px;
      margin-bottom: 16px;
      position: fixed;
      top: 0;
      left: 0;
      right: 0;
      background-color: white;
      z-index: 2;
      &:after {
        content: "";
        position: absolute;
        left: 0;
        width: 110%;
        height: 1px;
        background-color: $global-border;
        bottom: 0;
        margin: 0 -16px;
      }
    }
    h1,
    h2,
    h3,
    h4,
    h5 {
      margin-bottom: 16px;
      @media (max-width: $breakpoint-small-max) {
        font-size: 14px;
        margin: 0;
      }
    }
  }

  &__item {
    > * {
      margin: 8px 0;
    }
    padding: 6px 0;
    position: relative;
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin-top: 0px;
      margin-bottom: 4px;
    }
  }
  &__footer {
    display: flex;
    justify-content: center;
    position: fixed;
    bottom: 0;
    left: 0;
    right: 0;
    padding: 16px;
    background-color: white;
    &:after {
      content: "";
      height: 1px;
      background-color: $global-border;
      width: 110%;
      top: 0;
      margin-left: -16px;
      position: absolute;
      left: 0;
    }
  }
}

.isActive {
  display: block;
  position: fixed;
  background-color: white;
  left: 0;
  right: 0;
  top: 30px;
  bottom: 60px;
  z-index: 2;
  overflow-y: scroll;
  overflow-x: hidden;
}
