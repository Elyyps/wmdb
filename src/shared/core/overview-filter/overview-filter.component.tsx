import * as React from "react";
import InputRange from "react-input-range";
import { Checkbox } from "../checkbox/checkbox.component";
import { ICheckboxUnique } from "@app/api/core/checkbox";
import styles from "./overview-filter-component.module.scss";
import Search from "@assets/icons/search.svg";
import { Input } from "../input";
import { Button } from "../button";
import CROSS from "@assets/icons/cross.svg";
import { IconComponent } from "@app/core/icon";
import { IOverviewFilterCategoryItem, IOverviewFilterItem } from "@app/api/modules/overview";
import { arrayRemoveElement } from "@app/util/array";
import { LinkComponent } from "../link";
import { CalendarComponent } from "../calendar";

export interface IOverviewFilterComponentProps {
  checkboxCount?: number;
  checkedItems?: any;
  clearFilter?: () => void;
  currentFilter?: IOverviewFilterItem;
  date?: any;
  dateOnchange?: any;
  filterItems?: IOverviewFilterCategoryItem[];
  filterText?: string;
  onFilterChange: (filterOptions: IOverviewFilterItem) => void;
  range?: number;
  rangeMax?: number;
  searchPlaceholder: string;
  sidebarList?: any;
}

const OverviewFilterComponent = (props: IOverviewFilterComponentProps) => {
  const [isActive, setIsActive] = React.useState(false);
  const [checkedItems, setCheckedItems] = React.useState<ICheckboxUnique[]>([]);
  const [range, setRange] = React.useState<any>(null);
  const [tempRange, setTempRange] = React.useState<any>(null);
  const [filterText, setFilterText] = React.useState("");
  const [keywordText, setKeywordText] = React.useState("");
  const MEDIUM_BREAKPOINT = 960;
  const {
    onFilterChange,
    searchPlaceholder,
    rangeMax,
    filterItems,
    dateOnchange,
    date,
    sidebarList,
    currentFilter
  } = props;

  const clearFilter = () => {
    setCheckedItems([]);
    setFilterText("");
    setKeywordText("");
    setRange(0);
  };
  const initializeFilter = () => {
    if (currentFilter) {
      setCheckedItems(currentFilter.checkedItems);
      setRange(currentFilter.range);
      setKeywordText(currentFilter.keyword);
      setFilterText(currentFilter.filterText);
    }
  };
  const handleClick = () => {
    setIsActive(!isActive);
    initializeFilter();
  };

  const sendFilterOptions = (filter: IOverviewFilterItem) => {
    if (onFilterChange) {
      if (window.outerWidth >= MEDIUM_BREAKPOINT) {
        onFilterChange(filter);
      }
    }
  };

  const handleCheckboxChange = (item: ICheckboxUnique) => {
    let newCheckedItems = [...checkedItems];
    const currentItem = newCheckedItems.find(filter => filter.id === item.id);
    if (currentItem) {
      newCheckedItems = arrayRemoveElement(newCheckedItems, currentItem);
    } else {
      newCheckedItems.push({
        isChecked: true,
        id: item.id,
        label: item.label
      });
    }
    setCheckedItems(newCheckedItems);
    sendFilterOptions({ checkedItems: newCheckedItems, keyword: keywordText, filterText, range });
  };

  React.useEffect(() => {
    if (isActive) {
      document.documentElement.style.overflow = "hidden";
      document.body.style.overflowY = "scroll";
      initializeFilter();
    }

    return () => {
      document.documentElement.style.overflow = "auto";
      document.body.style.overflow = "auto";
      clearFilter();
    };
  }, [isActive]);

  React.useEffect(() => {
    initializeFilter();
  }, []);

  React.useEffect(() => {
    initializeFilter();
  }, [currentFilter, isActive]);

  return (
    <div className={styles["overview-filter-wrapper"]}>
      <button className={`${styles["button-search"]} uk-hidden@m`} onClick={handleClick}>
        <IconComponent icon={Search} size={"15px"} />
        <span>{searchPlaceholder}</span>
      </button>

      <div className={`${styles["overview-filter"]} ${isActive && styles["isActive"]}`}>
        {isActive ? (
          <div className={styles["overview-filter__title"]}>
            <button role={"button"} onClick={handleClick} className="uk-hidden@m">
              <IconComponent icon={CROSS} size={"15px"} />
            </button>
            <h2>Filters</h2>
            <span
              onClick={() => {
                clearFilter();
              }}
              role="button"
              className={"uk-hidden@m"}
            >
              Wissen
            </span>
          </div>
        ) : (
          <div className={styles["overview-filter__header"]}>
            <h2>Filters</h2>
            <Input
              value={filterText}
              onChange={(e: any) => {
                setFilterText(e.target.value);
                sendFilterOptions({ checkedItems, keyword: keywordText, filterText: e.target.value, range });
              }}
              placeholder={searchPlaceholder}
              icon={Search}
              name={"search"}
            />
          </div>
        )}
        {isActive && (
          <Input
            onChange={(e: any) => {
              setFilterText(e.target.value);
              sendFilterOptions({ checkedItems, keyword: keywordText, filterText: e.target.value, range });
            }}
            value={filterText}
            placeholder={searchPlaceholder}
            icon={Search}
            name={"search"}
          />
        )}
        {filterItems &&
          filterItems.map((filterItem, key) => (
            <div key={key} className={styles["overview-filter__item"]}>
              <h5>{filterItem.title}</h5>
              {filterItem.checkboxes.map((item: ICheckboxUnique, itemKey: any) => (
                <Checkbox
                  key={itemKey}
                  isChecked={checkedItems.find(filter => filter.id === item.id)}
                  id={key}
                  label={item.label}
                  name={item.id}
                  value={item.label}
                  count={item.count}
                  onChange={() => {
                    handleCheckboxChange(item);
                  }}
                />
              ))}
            </div>
          ))}

        <div className={styles["overview-filter__item"]}>
          <h5>Zoekwoord</h5>
          <Input
            value={keywordText}
            onChange={(e: any) => {
              setKeywordText(e.target.value);
              sendFilterOptions({ checkedItems, keyword: e.target.value, filterText, range });
            }}
            name="zoekwoord"
            placeholder="Zoekwoord"
          />
        </div>
        {rangeMax && (
          <div className={styles["overview-filter__item"]}>
            <h5>Personen</h5>
            <div className="input-range__wrapper">
              <InputRange
                maxValue={rangeMax}
                minValue={0}
                value={tempRange ? tempRange : 0}
                onChangeComplete={event => {
                  const numberRange = parseInt(event.toString(), 0);
                  setRange(numberRange);
                  sendFilterOptions({
                    checkedItems,
                    keyword: keywordText,
                    filterText,
                    range: numberRange
                  });
                }}
                onChange={event => {
                  setTempRange(event);
                }}
              />
              <div className="input-range__items">
                <span>{tempRange ? tempRange : 0}+</span>
                <span>{rangeMax}+</span>
              </div>
            </div>
          </div>
        )}
        {date && (
          <div className={styles["overview-filter__item"]}>
            <CalendarComponent onChange={dateOnchange} date={date} />
          </div>
        )}

        {sidebarList && (
          <div className={`${styles["overview-filter__item"]} ${styles["divider"]}`}>
            <h5>{sidebarList.title}</h5>
            <ul className={"sidebar-list"}>
              {sidebarList.list &&
                sidebarList.list.map((item: string, index: number) => (
                  <li key={index}>
                    <LinkComponent variant="secondary" to={item}>
                      {item}
                    </LinkComponent>
                  </li>
                ))}
            </ul>
          </div>
        )}
        <div className={`${styles["overview-filter__footer"]} uk-hidden@m`}>
          <Button
            onClick={() => {
              if (onFilterChange) {
                onFilterChange({ checkedItems, keyword: keywordText, filterText, range });
              }
              setIsActive(false);
            }}
            fullWidth
            title={"Toon uitjes"}
            variant={"primary"}
          />
        </div>
      </div>
    </div>
  );
};

export { OverviewFilterComponent };
