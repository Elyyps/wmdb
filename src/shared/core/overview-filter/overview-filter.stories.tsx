import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { OverviewFilterComponent } from "./overview-filter.component";
import { dummyOverviewFilterData } from "@app/api/core/overview-filter";

storiesOf("OverviewFilter", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <OverviewFilterComponent
      searchPlaceholder="Plaats, regio of provincie"
      range={0}
      onFilterChange={jest.fn()}
      {...dummyOverviewFilterData}
    />
  ));
