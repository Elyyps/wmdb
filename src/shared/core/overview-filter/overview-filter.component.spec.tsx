import React from "react";
import { shallow } from "enzyme";
import { OverviewFilterComponent } from "./overview-filter.component";
import { dummyOverviewFilterData } from "@app/api/core/overview-filter";
import { getCardsPaginated } from "@app/api/modules/overview";
// tslint:disable
describe("[OverviewFilter]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <OverviewFilterComponent
        searchPlaceholder="Plaats, regio of provincie"
        range={0}
        onFilterChange={jest.fn()}
        {...dummyOverviewFilterData}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it("should return 200", () => {
    const cards200 = getCardsPaginated(0, 0, { filterText: "", range: 0, checkedItems: [], keyword: "" });
    expect(cards200.total).toBe(200);
  });
  it("should return 200", () => {
    const cards200 = getCardsPaginated(10, 0, { filterText: "", range: 0, checkedItems: [], keyword: "" });
    expect(cards200.total).toBe(200);
  });
  it("should return 0", () => {
    const cards200 = getCardsPaginated(200, 0, { filterText: "", range: 0, checkedItems: [], keyword: "" });
    expect(cards200.total).toBe(200);
  });
  it("should return 1", () => {
    const cards1 = getCardsPaginated(0, 0, { filterText: "", range: 0, checkedItems: [], keyword: "keyword" });
    expect(cards1.total).toBe(1);
  });
  it("should return 0", () => {
    const cards1 = getCardsPaginated(0, 0, {
      filterText: "",
      range: 0,
      checkedItems: [{ id: 1, label: "" }],
      keyword: "keyword"
    });
    expect(cards1.total).toBe(0);
  });
  it("should return 84", () => {
    const cards1 = getCardsPaginated(0, 0, {
      filterText: "",
      range: 0,
      checkedItems: [{ isChecked: true, id: 1, label: "" }, { isChecked: true, id: 2, label: "" }],
      keyword: ""
    });
    expect(cards1.total).toBe(84);
  });
  it("should return 129", () => {
    const cards1 = getCardsPaginated(0, 0, {
      filterText: "",
      range: 0,
      checkedItems: [
        { isChecked: true, id: 1, label: "" },
        { isChecked: true, id: 2, label: "" },
        { isChecked: true, id: 3, label: "" },
        { isChecked: true, id: 4, label: "" }
      ],
      keyword: ""
    });
    expect(cards1.total).toBe(129);
  });
});
