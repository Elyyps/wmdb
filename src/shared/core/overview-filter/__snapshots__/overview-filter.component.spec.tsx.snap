// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[OverviewFilter] should render component without crashing 1`] = `
<div
  className="overview-filter-wrapper"
>
  <button
    className="button-search uk-hidden@m"
    onClick={[Function]}
  >
    <div>
      <IconComponent
        icon="search.svg"
        size="15px"
      />
      <span>
        Plaats, regio of provincie
      </span>
    </div>
    <div
      className="button-search-count"
    >
      <span>
        0
      </span>
    </div>
  </button>
  <div
    className="overview-filter false"
  >
    <div
      className="overview-filter__header"
    >
      <h2>
        Filters
      </h2>
      <Input
        icon="search.svg"
        name="search"
        onChange={[Function]}
        placeholder="Plaats, regio of provincie"
        value=""
      />
    </div>
    <br />
    <div
      className="overview-filter__item"
      key="0"
    >
      <h5>
        Spel & Ontspanning
      </h5>
      <Checkbox
        id={0}
        key="0"
        label="Citygames"
        name={1}
        onChange={[Function]}
        value="Citygames"
      />
      <Checkbox
        id={0}
        key="1"
        label="Beachgames"
        name={2}
        onChange={[Function]}
        value="Beachgames"
      />
      <Checkbox
        id={0}
        key="2"
        label="Indoor Spel en Ontspanning"
        name={3}
        onChange={[Function]}
        value="Indoor Spel en Ontspanning"
      />
      <Checkbox
        id={0}
        key="3"
        label="Outdoor Spel en Ontspanning"
        name={4}
        onChange={[Function]}
        value="Outdoor Spel en Ontspanning"
      />
    </div>
    <div
      className="overview-filter__item"
      key="1"
    >
      <h5>
        Sportief & Actie(f)
      </h5>
      <Checkbox
        id={1}
        key="0"
        label="Fun op Wielen"
        name={5}
        onChange={[Function]}
        value="Fun op Wielen"
      />
      <Checkbox
        id={1}
        key="1"
        label="Outdoor Lucht"
        name={6}
        onChange={[Function]}
        value="Outdoor Lucht"
      />
      <Checkbox
        id={1}
        key="2"
        label="Outdoor Water"
        name={7}
        onChange={[Function]}
        value="Outdoor Water"
      />
      <Checkbox
        id={1}
        key="3"
        label="Indoor Sport en Actie"
        name={8}
        onChange={[Function]}
        value="Indoor Sport en Actie"
      />
      <Checkbox
        id={1}
        key="4"
        label="Outdoor Sport en Actie"
        name={26}
        onChange={[Function]}
        value="Outdoor Sport en Actie"
      />
    </div>
    <div
      className="overview-filter__item"
      key="2"
    >
      <h5>
        Creatieve Uitjes
      </h5>
      <Checkbox
        id={2}
        key="0"
        label="Creatief"
        name={9}
        onChange={[Function]}
        value="Creatief"
      />
      <Checkbox
        id={2}
        key="1"
        label="Theater"
        name={10}
        onChange={[Function]}
        value="Theater"
      />
      <Checkbox
        id={2}
        key="2"
        label="Muziek en Dans"
        name={11}
        onChange={[Function]}
        value="Muziek en Dans"
      />
    </div>
    <div
      className="overview-filter__item"
      key="3"
    >
      <h5>
        Workshops
      </h5>
      <Checkbox
        id={3}
        key="0"
        label="Kookworkshops"
        name={12}
        onChange={[Function]}
        value="Kookworkshops"
      />
      <Checkbox
        id={3}
        key="1"
        label="Kunst en Cultuur"
        name={13}
        onChange={[Function]}
        value="Kunst en Cultuur"
      />
      <Checkbox
        id={3}
        key="2"
        label="Body en Spirit"
        name={14}
        onChange={[Function]}
        value="Body en Spirit"
      />
    </div>
    <div
      className="overview-filter__item"
      key="4"
    >
      <h5>
        Eten, Drinken & Feesten
      </h5>
      <Checkbox
        id={4}
        key="0"
        label="Eten en drinken"
        name={15}
        onChange={[Function]}
        value="Eten en drinken"
      />
      <Checkbox
        id={4}
        key="1"
        label="Themafeesten"
        name={16}
        onChange={[Function]}
        value="Themafeesten"
      />
      <Checkbox
        id={4}
        key="2"
        label="Totaalprogramma"
        name={17}
        onChange={[Function]}
        value="Totaalprogramma"
      />
      <Checkbox
        id={4}
        key="3"
        label="Attracties en Entertainment"
        name={18}
        onChange={[Function]}
        value="Attracties en Entertainment"
      />
    </div>
    <div
      className="overview-filter__item"
      key="5"
    >
      <h5>
        Teambuilden
      </h5>
      <Checkbox
        id={5}
        key="0"
        label="Teamtraining"
        name={19}
        onChange={[Function]}
        value="Teamtraining"
      />
      <Checkbox
        id={5}
        key="1"
        label="Teambuilding"
        name={20}
        onChange={[Function]}
        value="Teambuilding"
      />
      <Checkbox
        id={5}
        key="2"
        label="Uit Vergaderen"
        name={21}
        onChange={[Function]}
        value="Uit Vergaderen"
      />
    </div>
    <div
      className="overview-filter__item"
      key="6"
    >
      <h5>
        Incentives
      </h5>
      <Checkbox
        id={6}
        key="0"
        label="Wintersport"
        name={22}
        onChange={[Function]}
        value="Wintersport"
      />
      <Checkbox
        id={6}
        key="1"
        label="Stedentrips"
        name={23}
        onChange={[Function]}
        value="Stedentrips"
      />
      <Checkbox
        id={6}
        key="2"
        label="Adventure & Outdoor"
        name={24}
        onChange={[Function]}
        value="Adventure & Outdoor"
      />
      <Checkbox
        id={6}
        key="3"
        label="Reizen Buitenland"
        name={25}
        onChange={[Function]}
        value="Reizen Buitenland"
      />
    </div>
    <div
      className="overview-filter__item uk-visible@m"
    >
      <h5>
        Zoekwoord
      </h5>
      <Input
        name="zoekwoord"
        onChangeText={[Function]}
        placeholder="Zoekwoord"
        value=""
      />
    </div>
    <div
      className="overview-filter__item"
    >
      <h5>
        Personen 
      </h5>
      <div
        className="input-range__wrapper"
      >
        <InputRange
          allowSameValues={false}
          classNames={
            Object {
              "activeTrack": "input-range__track input-range__track--active",
              "disabledInputRange": "input-range input-range--disabled",
              "inputRange": "input-range",
              "labelContainer": "input-range__label-container",
              "maxLabel": "input-range__label input-range__label--max",
              "minLabel": "input-range__label input-range__label--min",
              "slider": "input-range__slider",
              "sliderContainer": "input-range__slider-container",
              "track": "input-range__track input-range__track--background",
              "valueLabel": "input-range__label input-range__label--value",
            }
          }
          disabled={false}
          maxValue={100}
          minValue={0}
          onChange={[Function]}
          onChangeComplete={[Function]}
          step={1}
          value={null}
        />
        <div
          className="input-range__items"
        >
          <span>
            0
          </span>
          <span>
            100
            +
          </span>
        </div>
      </div>
    </div>
    <div
      className="overview-filter__footer uk-hidden@m"
    >
      <Button
        fullWidth={true}
        onClick={[Function]}
        title="Toon uitjes"
        variant="primary"
      />
    </div>
  </div>
</div>
`;
